bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -0,0 +1,441 @@
+        final Console console = new JRubyConsole(getScope(), out, session);
+
+        if (args.length > 0) {
+            String command = args[0];
+            switch (command) {
+                case INFO:
+                    info(console);
+                    break;
+                case CONSOLE:
+                    startConsole(console);
+                    break;
+                case GEM:
+                    gem(console, Arrays.stream(args).skip(1).toList());
+                    break;
+                case UPDATE:
+                    updateGems(console);
+                    break;
+                case PRUNE:
+                    if (args.length > 1) {
+                        if ("-f".equals(args[1]) || "--force".equals(args[1])) {
+                            cleanupOtherGemHomes(console, true);
+                        } else {
+                            console.println("Use -f or --force to skip confirmation.");
+                        }
+                    } else {
+                        cleanupOtherGemHomes(console, false);
+                    }
+                    break;
+                case "--help":
+                    printUsage(console);
+                    break;
+                default:
+                    console.println("Unknown command '" + command + "'");
+                    printUsage(console);
+                    break;
+            }
+        } else {
+            printUsage(console);
+        }
+        return null;
+    }
+
+    private void info(Console console) {
+        final String PRINT_VERSION_NUMBERS = """
+                library_version = defined?(OpenHAB::DSL::VERSION) && OpenHAB::DSL::VERSION
+
+                puts "JRuby #{JRUBY_VERSION}"
+                puts "JRuby Scripting Library #{library_version || 'is not installed'}"
+                puts "GEM_HOME: #{ENV['GEM_HOME']}"
+                puts "RUBYLIB: #{ENV['RUBYLIB']}"
+                    """;
+
+        executeWithFullJRuby(console, engine -> engine.eval(PRINT_VERSION_NUMBERS));
+        console.println("Script path: " + scriptFileWatcher.getWatchPath());
+        console.println("");
+        console.println("JRuby Scripting Add-on Configuration:");
+        console.println("=====================================");
+        jRubyScriptEngineFactory.getConfiguration().getConfigurations().forEach((key, value) -> {
+            console.println(key + ": " + value);
+        });
+    }
+
+    private void startConsole(Console console) {
+        final String consoleScript = jRubyScriptEngineFactory.getConfiguration().getConsole();
+
+        if (consoleScript.isBlank()) {
+            console.println(
+                    "No console script configured. Please set the 'console_script' property in the configuration.");
+            return;
+        }
+
+        executeWithFullJRuby(console, engine -> {
+            engine.getContext().setAttribute("$console", console, ScriptContext.ENGINE_SCOPE);
```

Comment:
do we really need console? or would session be enough? (really, all I'm using right now is terminal - which you get from session). then you don't event need the JRubyConsole. in the one or two places that need session, just pass it through from execute().

Reply:
I've changed it to just $terminal. However, we still need `JRubyConsole` because the OSGiConsole class from core is in an internal package. We still want to have a `console` object to keep a resemblance to the standard ConsoleCommandExtension in case we need to refactor this back if https://github.com/openhab/openhab-core/pull/4635 is accepted.